src/client/dfs/dfs.c

```diff
@@ -1476,6 +1480,52 @@ dfs_cont_create2(daos_handle_t poh, uuid_t *cuuid, dfs_attr_t *attr, daos_handle
 	return dfs_cont_create_int(poh, cuuid, false, NULL, attr, coh, dfs);
 }
 
+int
+dfs_cont_create_with_label(daos_handle_t poh, const char *label, dfs_attr_t *attr,
+			   uuid_t *cuuid, daos_handle_t *coh, dfs_t **dfs)
+{
+	daos_prop_t		*label_prop;
+	daos_prop_t		*merged_props = NULL;
+	daos_prop_t		*orig = NULL;
+	dfs_attr_t		local = {};
+	int			rc;
+
+	label_prop = daos_prop_alloc(1);
+	if (label_prop == NULL)
+		return ENOMEM;
+
+	label_prop->dpp_entries[0].dpe_type = DAOS_PROP_CO_LABEL;
+	rc = daos_prop_entry_set_str(&label_prop->dpp_entries[0], label, DAOS_PROP_LABEL_MAX_LEN);
+	if (rc)
+		D_GOTO(out_prop, rc = daos_der2errno(rc));
+
+	if (attr == NULL)
+		attr = &local;
+
+	if (attr->da_props) {
+		merged_props = daos_prop_merge(attr->da_props, label_prop);
+		if (merged_props == NULL)
+			D_GOTO(out_prop, rc = ENOMEM);
+		orig = attr->da_props;
+		attr->da_props = merged_props;
+	} else {
+		attr->da_props = label_prop;
+	}
+
+	if (cuuid == NULL) {
+		uuid_t u;
+
+		rc = dfs_cont_create_int(poh, &u, false, NULL, attr, coh, dfs);
```

Comment:
Seems like the uuid is thrown away here, is that intentional?

Reply:
yes, this is when a user creates with label, but does not pass a sink buffer for the uuid, as they don't care about using it.